migration-to-python-3-only.md

```diff
@@ -0,0 +1,207 @@
+the same time is significant, and slow the Jupyter and IPython open-source teams
+to achieve many of our goals.
+
+We also believe that the maturity of Python 3 is sufficient enough, and a
+sufficient number of project in the scientific community are ready to make the
+commitment to drop Python 2 support.
+
+We want to provide a seamless experience to current Python 3 users without
+breaking any workflow and pipeline for Python 2 users. Thus we want in a
+relatively short period of time to remove the burden of support ting Python 2
+support from the core team, while preserving functionality for users. 
+
+We also recognize that companies and laboratories may rely on Python 2, and
+want to offer solution for these too, whether they directly use, or sell
+product relying on Jupyter and IPython.
+
+# IPython 5.x LTS
+
+This part concern the reference kernel implementation of Jupyter, also know as
+IPython. This specific kernel implementation also rely on `ipykernel`, `traitlets`
+and `ipython_genutils`.
+
+We propose that the next major version cycle (5.x) of IPython be the last
+version to support Python 2. As we recognize that Python 2 is supported by the
+CPython core team until 2020, we propose to make the 5.x major cycle a Long
+Term Support version under the following intent:
+
+  - For the 18 month following the release of IPython 5.0, the core team will
+    engage in critical & security bug fix for the 5.x branch, as well as
+    regular release (both of minor and patch revision) of IPython 5.x branch.
+    This apply also to the documentation of the 5.x branch. THe core team will
+    also accept to provide reasonable support requests for the IPython 5.x
+    branch.
+
+
+  - For the 36 month following the release of IPython 5.x, the core team will
+    continue to review critical & security releated patches, as well as
+    improvment to the documentation, and make related release if necessary. The
+    core team will make sure that the 5.x Version of IPython stay visible
+    various web pages mentionning how to get IPython. The core team reserve the
+    right to ignore and or close any support request, request for patch
+    integration or other communication that oes not go into the previous
+    category.
+
+  - The IPython/Jupyter team will do a best effort to provide fixes in the
+    realm of reasonable requests, and will actively try to provide upstream
+    patches if IPython missbehavior is due dependedncies, unless otherwise
+    specified. I.E, we will backport and release fix to traitlets, and
+    ipykernel only if it dirrectly affect IPython
+
+  - pass 36 month after IPython 5.0 release, IPython 5.x will be considers
+    unsupported.
+
+  - The above time lines are intent, and the IPython core team reserve the
+    right to modify them in the future.
+
+  - regardless of above statement, Python 2 support in the 5.x branch will not
+    exceed support of CPython 2.7, and be likely much shorter.
```

Comment:
It's not likely to be *much* shorter - the timeline you proposed only has us dropping support for 2.7 a year before Python itself does.